src/vss_tools/vspec/vssexporters/vss2apigear.py

```diff
@@ -0,0 +1,501 @@
+            file.write(f"      - name: {prop_key}\n")
+            if prop_value.description is not None:
+                file.write(f"        description: \"{prop_value.description}\"\n")
+            file.write(f"        type: {prop_value.type.type}\n")
+            if prop_value.type.is_array:
+                file.write("        array: true\n")
+
+    if len(module.enumerations) > 0:
+        file.write("\n")
+        file.write("enums:\n")
+
+        for enumeration in module.enumerations:
+            file.write(f"  - name: {enumeration.name}\n")
+            file.write("    members:\n")
+
+            for variant in enumeration.variants:
+                file.write(f"      - name: {variant}\n")
+
+    if len(module.structures) > 0:
+        file.write("\n")
+        file.write("structs:\n")
+
+        for structure in module.structures:
+            file.write(f"  - name: {structure.name}\n")
+            file.write("    fields:\n")
+
+            for prop_key, prop_value in structure.properties.items():
+                file.write(f"      - name: {prop_key}\n")
+                if prop_value.description is not None:
+                    file.write(f"        description: \"{prop_value.description}\"\n")
+                file.write(f"        type: {prop_value.type.type}\n")
+                if prop_value.type.is_array:
+                    file.write("        array: true\n")
+
+
+def export_apigear(directory, root, data_type_tree, layers: typing.Dict[SolutionLayers, str]):
+    """This method is used to traverse through the root VSS node to build
+       -> ApiGear equivalent string buffer and to serialize it accordingly into a file
+    """
+    module = ApiGearModule()
+    module_name = root.name
+    module_filename = MODULE_FILENAME.format(module_name)
+    module_path = os.path.join(directory, module_filename)
+
+    log.info(f"directory: {directory}")
+    log.info(f"Module name: {module_name}")
+    log.info(f"Module filename: {module_filename}")
+    log.info(f"Module Path: {module_path}")
+
+    if (not layers):
+        layers = {SolutionLayers.CPP: "../cppservice"}
+        log.warning(f"No layers provided! Defaulting to CPP in {layers[SolutionLayers.CPP]}")
+
+    log.debug(layers)
+
+    export_node(root, module)
+    export_data_type_node(data_type_tree, module)
+    generate_solution(directory, module_filename, module_name, layers)
+    with open(module_path, "w") as file:
+        generate_module(file, module, module_name)
+        log.info(f"Generated module file at {module_path}")
+
+    log.info("Apigear files generated at location : " + file.name)
+
+
+@click.command()
+@clo.vspec_opt
+@clo.output_required_opt
+@clo.include_dirs_opt
+@clo.extended_attributes_opt
+@clo.strict_opt
+@clo.aborts_opt
+@clo.uuid_opt
+@clo.expand_opt
+@clo.overlays_opt
+@clo.quantities_opt
+@clo.units_opt
+@clo.types_opt
+@clo.types_output_opt
+@click.option(
+    "--apigear-template-unreal-path",
```

Comment:
If it is a file then have a look at `cli_options.py` how to specify that to make it a `pathlib` `Path` and checking whether it exists and can be read. `is_flag=false` is not required. `required=False` is also the default and can be skipped

Reply:
Paths from arguments are passed to the ApiGar which is responsible for their creation and ownership, the exporter itself doesn't actually rely on those.

Reply:
Changed the path options types to `Path` as you suggested, and removed unnecessary `is_flag=False` and `required=False`.